### ISSUE TITLE
[Java] Add utilities to (de)serialize Descriptors

### ISSUE THREAD

**Issue by alexvanboxel** [protocolbuffers/protobuf#6101]
**What language does this apply to?**
Java

I have already 3 different project where I needed to (de)serialize. Not willing to copy/paste the code between the different open source project I like to donate the utilities that help to (de)serialize them in standard Serializable object.

PR will follow.

**Comment by haon4** [comment]
Thanks for offering to donate the utilities! Can you clarify a bit on what you are trying to (de)serialize? All Descriptors are already proto Message that can be serialize/deserialize already. What would the utility do on top of that?

**Comment by alexvanboxel** [comment]
I mostly bump into Java serialization problems in my Beam pipeline. The graph is expected to be (de)serialable, but in a Java way and that means I need to get the Descriptor serializer (in a Java way). I've already used it 2x.

Here: Currenly in a PR (on Apache Beam):
https://github.com/apache/beam/pull/8496/files#diff-a059af3085838528852db6f97b17602e

And in another project:
https://github.com/anemos-io/protobeam/blob/master/convert/src/main/java/io/anemos/protobeam/convert/DescriptorSerializer.java

basically it would end up in the java util package (as it is a re-usable) utility to make sure that we can Java (de)serialize. It does this by putting on the Object Stream:

1) Full file name
2) Message name
3) Full FileDescriptorSet (included any referenced dependencies)

WDYT?

**Comment by haon4** [comment]
I would leave the (de)serialize to ObjectInputStream/ObjectOutputStream out of this. Having a general utility to create a FileDescriptorSet from a FileDescriptor seems generally useful. Similarly, a utility that find a FileDescriptor from a FileDescriptorSet by name seems useful too. However, I would leave it to the application to decide what to do with FileDescriptorSet. They can (de)serialize it under their own way, or do something else with it.

**Comment by alexvanboxel** [comment]
Ok, makes sense. With the utils the implementation that the user needs to write is then much smaller. I'll see if I'll make a PR tomorow.
